driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -182,6 +188,23 @@ public RequestHandler(
     } else if (statement instanceof BoundStatement) {
       statementType = "prepared";
       statementText = ((BoundStatement) statement).statement.getQueryString();
+      keyspace = ((BoundStatement) statement).getKeyspace();
+
+      ColumnDefinitions boundColumns =
+          ((BoundStatement) statement).statement.getPreparedId().boundValuesMetadata.variables;
+
+      StringBuilder partitionKeyBuilder = new StringBuilder(PARTITION_KEY_MAX_LENGTH);
+      for (int i = 0; i < boundColumns.size(); i++) {
```

Comment:
Maybe the example I sent was misleading, but the code should not iterate over all bound values (you can bind a value to a column not part of the primary/partition key). Instead it should iterate over only partition key columns - those are alternatively called routing key columns (the partition key determines where to route the request).

The fix is easy - do not iterate over all `boundColumns`, just those specified in `((BoundStatement) statement).statement.getPreparedId().routingKeyIndexes`, i.e.:

``` java
for (int i : ((BoundStatement) statement).statement.getPreparedId().routingKeyIndexes) {
```